no-undefined-types gives incorrect warnings since 51.0.5

## Expected behavior
I expect no warnings when using a defined type. The same code worked fine in v51.0.4 and earlier.

## Actual behavior
I get the following warnings:
```
test.js
  1:7  warning  'otherFile' is assigned a value but never used  no-unused-vars
  5:1  warning  The type 'otherFile.MyType' is undefined        jsdoc/no-undefined-types
```

## ESLint Config
```js
const globals = require('globals');
const js = require('@eslint/js');
const jsdoc = require('eslint-plugin-jsdoc');

module.exports = [
  js.configs.recommended,
  {
    plugins: {
      jsdoc,
    },
    languageOptions: {
      globals: {
        ...globals.node,
      },
    },
    rules: {
      'jsdoc/no-undefined-types': 1,
      'no-unused-vars': ['warn', { args: 'none' }],
    },
  },
];
```

## Example
other-file.js
```js
/**
 * @typedef {Object} MyType
 */
```

test.js
```js
const otherFile = require('./other-file');

/**
 * A function
 * @param {otherFile.MyType} a
 */
function f(a) {}

f();
```

## Environment

- Node version: 22
- ESLint version 9.30.0
- `eslint-plugin-jsdoc` version: 51.0.5 - 51.3.3

--- a/docs/rules/no-undefined-types.md
+++ b/docs/rules/no-undefined-types.md
@@ -996,5 +996,13 @@ class SomeClass {
  *
  * @member {Intl.Collator}
  */
+
+const otherFile = require('./other-file');
+
+/**
+ * A function
+ * @param {otherFile.MyType} a
+ */
+function f(a) {}
 ````
 

--- a/src/rules/noUndefinedTypes.js
+++ b/src/rules/noUndefinedTypes.js
@@ -312,6 +312,19 @@ export default iterateJsdoc(({
                   return `${name}.${property}`;
                 }).filter(Boolean),
               ];
+            case 'VariableDeclarator':
+              if (/** @type {import('@typescript-eslint/types').TSESTree.Identifier} */ (
+                /** @type {import('@typescript-eslint/types').TSESTree.CallExpression} */ (
+                  globalItem?.init
+                )?.callee)?.name === 'require'
+              ) {
+                imports.push(/** @type {import('@typescript-eslint/types').TSESTree.Identifier} */ (
+                  globalItem.id
+                ).name);
+                break;
+              }
+
+              return [];
           }
 
           return [

--- a/test/rules/assertions/noUndefinedTypes.js
+++ b/test/rules/assertions/noUndefinedTypes.js
@@ -1729,5 +1729,16 @@ export default /** @type {import('../index.js').TestCases} */ ({
          */
       `,
     },
+    {
+      code: `
+        const otherFile = require('./other-file');
+
+        /**
+         * A function
+         * @param {otherFile.MyType} a
+         */
+        function f(a) {}
+      `,
+    },
   ],
 });

